packages/salesforcedx-vscode-core/src/commands/stopApexDebugLogging.ts

```diff
@@ -1,90 +1,63 @@
+  ChannelService,
+  TraceFlagsRemover
 } from '@salesforce/salesforcedx-utils-vscode';
 import * as vscode from 'vscode';
-import { hideTraceFlagExpiration } from '../decorators';
-import { nls } from '../messages';
+import { OUTPUT_CHANNEL } from '../channels';
+import { WorkspaceContext } from '../context';
 import { telemetryService } from '../telemetry';
 import { developerLogTraceFlag } from '.';
-import { SfCommandlet, SfCommandletExecutor, SfWorkspaceChecker } from './util';
+import { handleStartCommand, handleFinishCommand } from '../utils/channelUtils';
 
-export class StopApexDebugLoggingExecutor extends SfCommandletExecutor<{}> {
-  public build(): Command {
-    return deleteTraceFlag();
-  }
-
-  public execute(response: ContinueResponse<{}>): void {
-    const startTime = process.hrtime();
-    const cancellationTokenSource = new vscode.CancellationTokenSource();
-    const cancellationToken = cancellationTokenSource.token;
-
-    const execution = new CliCommandExecutor(this.build(), {
-      cwd: workspaceUtils.getRootWorkspacePath()
-    }).execute(cancellationToken);
-
-    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
-    execution.processExitSubject.subscribe(async data => {
-      this.logMetric(execution.command.logName, startTime);
-      if (data !== undefined && String(data) === '0') {
-        developerLogTraceFlag.turnOffLogging();
-        hideTraceFlagExpiration();
-      }
-    });
-  }
-}
+const command = 'SFDX: Turn Off Apex Debug Log for Replay Debugger';
 
 export const turnOffLogging = async (): Promise<void> => {
+  console.log('Enter turnOffLogging()');
+  const channelService = new ChannelService(OUTPUT_CHANNEL);
+  handleStartCommand(channelService, command);
+
   if (developerLogTraceFlag.isActive()) {
-    const execution = new CliCommandExecutor(deleteTraceFlag(), {
-      cwd: workspaceUtils.getRootWorkspacePath()
-    }).execute();
-    telemetryService.sendCommandEvent(execution.command.logName);
-    const resultPromise = new CommandOutput().getCmdResult(execution);
-    const result = await resultPromise;
-    const resultJson = JSON.parse(result);
-    if (resultJson.status === 0) {
-      return Promise.resolve();
-    } else {
-      return Promise.reject('Restoring the debug levels failed.');
-    }
-  }
-};
+    console.log('Developer log trace flag is active');
+    try {
+      await vscode.window.withProgress(
+        {
+          location: vscode.ProgressLocation.Notification,
+          title: command,
+          cancellable: false
+        },
+        async progress => {
+          progress.report({ message: 'Running...' });
+          const nonNullTraceFlag = developerLogTraceFlag.getTraceFlagId()!;
+          const connection = await WorkspaceContext.getInstance().getConnection();
+          await TraceFlagsRemover.getInstance(connection).removeTraceFlag(nonNullTraceFlag);
+          telemetryService.sendCommandEvent('stop_apex_debug_logging');
```

Comment:
do we think we'll probably always have telemetry in these?  Maybe the stuff you have in channelUtils should also have telemetry in it?